exec/run_test.go

```diff
@@ -41,3 +42,48 @@ func TestDefaultEnv(t *testing.T) {
 	})
 	assert.Contains(t, c.Env, "alice=bob")
 }
+
+func TestRun(t *testing.T) {
+	command := exec.Cmd{
+		Name: "pwd",
+	}
+	stdout, _, err := exec.Run(command)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, stdout)
+}
+
+func TestRunTimeout(t *testing.T) {
+	command := exec.Cmd{
+		Name:    "sleep",
+		Argv:    []string{"2"},
+		Timeout: 3 * time.Second,
+	}
+	_, _, err := exec.Run(command)
+	assert.NoError(t, err)
+
+	command.Timeout = 1 * time.Second
+	_, _, err = exec.Run(command)
+	assert.Contains(t, err.Error(), "timed out")
+}
+
+func TestRunRetry(t *testing.T) {
+	command := exec.Cmd{
+		Name:    "sleep",
+		Argv:    []string{"3"},
+		Timeout: 1 * time.Second,
+		Retries: 1,
+	}
+
+	start := time.Now()
+	_, _, err := exec.Run(command)
+	assert.Error(t, err)
+	assert.WithinDuration(t, start, time.Now(), 3*time.Second)
```

Comment:
I think it's a good idea to `assert.WithinDuration` as you do here, but the delta duration should be slightly _less_ than 3 seconds to make it clear that we don't want the "sleep" command to run to completion. Perhaps `2*time.Second + 500*time.Millisecond`

Reply:
Good call, I'm changing the sleep argument to `4` instead of modifying the time in the assertion.